desc/compute/_equil.py

```diff
@@ -231,6 +232,7 @@ def _J_sqrt_g_r(params, transforms, profiles, data, **kwargs):
     data=["J"],
 )
 def _J_R(params, transforms, profiles, data, **kwargs):
+    errorif(kwargs.get("basis", "rpz").lower() != "rpz", NotImplementedError)
```

Comment:
Instead of erroring in all these cases, couldn't we convert J (or whatever else) to rpz locally then get the component?

Reply:
No because the basis kwarg is not sufficient to specify that "J" is in xyz or rpz.
If user does `eq.compte(["J", "J_R"], basis="xyz")`,
then after the changes in this PR, (master would compute wrong), we would know that "J" is in "xyz" and we could do what you say.

If user passes in precomputed data  `eq.compte(["J_R"], basis="xyz", data={"J": J})` then we don't know whether the dependencies are in xyz or rpz.

Reply:
If the user is passing in data we are always going to have basis consistency issues, and I don't think there is an easy way to avoid that. We just have to trust the users know what they are doing.